src/main/java/fi/otavanopisto/kuntaapi/server/integrations/management/ManagementMenuEntityUpdater.java

```diff
@@ -145,43 +163,136 @@ private void updateManagementMenus(OrganizationId organizationId) {
+    MenuId kuntaApiMenuId = idController.translateMenuId(managementMenuId, KuntaApiConsts.IDENTIFIER_NAME);
+    if (kuntaApiMenuId == null) {
+      logger.info(String.format("Could not translate management menu %d into kunta api id", managementMenu.getId()));
+      return null;
+    }
+    
+    menu.setId(kuntaApiMenuId.getId());
+    menu.setSlug(managementMenu.getSlug());
+    
+    return menu;
+  }
+  
+  private MenuItem translateMenuItem(OrganizationId organizationId, fi.otavanopisto.mwp.client.model.Menuitem managementMenuItem) {
+    MenuItem menuItem = new MenuItem();
+    
+    MenuItemId managementMenuItemId = new MenuItemId(organizationId,ManagementConsts.IDENTIFIER_NAME, String.valueOf(managementMenuItem.getId()));
+    MenuItemId kuntaApiMenuItemId = idController.translateMenuItemId(managementMenuItemId, KuntaApiConsts.IDENTIFIER_NAME);
+    if (kuntaApiMenuItemId == null) {
+      logger.info(String.format("Could not translate management menu item %d into kunta api id", managementMenuItem.getId()));
+      return null;
+    }
+    
+    MenuItemType itemType = getItemType(managementMenuItem);
+    if (itemType == null) {
+      logger.severe(String.format("Could not determine item type for %d", managementMenuItem.getId()));
+      return null;
+    }
+    
+    PageId pageId = translatePageId(organizationId, managementMenuItem.getPageId());
+    MenuItemId parentMenuItemId = translateMenuItemId(organizationId, managementMenuItem.getParentItemId());
+    
+    menuItem.setId(kuntaApiMenuItemId.getId());
+    menuItem.setLabel(managementMenuItem.getTitle());
+    menuItem.setFileId(null);
+    menuItem.setExternalUrl(itemType == MenuItemType.LINK ? managementMenuItem.getUrl() : null);
+    menuItem.setPageId(pageId != null ? pageId.getId() : null);
+    menuItem.setParentItemId(parentMenuItemId != null ? parentMenuItemId.getId() : null);
+    menuItem.setType(itemType.toString());
+    
+    return menuItem;
+  }
+
+  private MenuItemId translateMenuItemId(OrganizationId organizationId, Long parentItemId) {
+    if (parentItemId == null) {
+      return null;
     }
     
-    modificationHashCache.put(identifier.getKuntaApiId(), createPojoHash(managementMenuItem));
+    MenuItemId managementMenuItem = new MenuItemId(organizationId, ManagementConsts.IDENTIFIER_NAME, String.valueOf(parentItemId));
+
+    return idController.translateMenuItemId(managementMenuItem, KuntaApiConsts.IDENTIFIER_NAME);
   }
 
+  private MenuItemType getItemType(fi.otavanopisto.mwp.client.model.Menuitem managementMenuItem) {
+    switch (managementMenuItem.getType()) {
+      case "page":
+        return MenuItemType.PAGE;
+      case "post":
+        return MenuItemType.NEWS_ARTICLE;
+      case "custom":
+        return MenuItemType.LINK;
+      default:
+        return null;
+    }
+  }
+  
+  private PageId translatePageId(OrganizationId organizationId, Long pageId) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "translatePageId" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)